### PR TITLE
Ensure special characters are escaped in TSCN connections

### LINKED ISSUE #86347: "\" used in node names prevents parsing of .TSCN
### Tested versions

Reproduce able in 4.3.stable

### System information

Windows 10 - vulkan Forward+ renderer, v4.3.stable

### Issue description

Initial Issue:
Currently in live Godot 4.3 branch, accidentally inserted the "\" key at the end of a Node name. While adding and implemented the node, I connected a signal, saved and ran the scene, causing an issue with it immediately crashing, saved and relaunched to see the file could not be parsed, at some point in the process the following offending line was added.
image

### Steps to reproduce

Rename a node to include \
example: new_button\
create a signal from the renamed node to any given node using the editor panel
save the scene
close and re-open scene to get parser error

### Minimal reproduction project (MRP)

[errorissue.zip](https://github.com/godotengine/godot/files/13721775/errorissue.zip)

### PR DESCRIPTION
**Pull request by LimestaX**
Added c_escape to signal connection write strings. Original issue #86347

_Bugsquad edit:_ Fixes #86347.

### CODE REVIEW

File: scene/resources/resource_format_text.cpp [hunk 1]
```diff
@@ -2289,10 +2289,10 @@ Error ResourceFormatSaverTextInstance::save(const String &p_path, const Ref<Reso
...
 
 			String connstr = "[connection";
-			connstr += " signal=\"" + String(state->get_connection_signal(i)) + "\"";
-			connstr += " from=\"" + String(state->get_connection_source(i).simplified()) + "\"";
-			connstr += " to=\"" + String(state->get_connection_target(i).simplified()) + "\"";
-			connstr += " method=\"" + String(state->get_connection_method(i)) + "\"";
+			connstr += " signal=\"" + String(state->get_connection_signal(i).c_escape()) + "\"";
+			connstr += " from=\"" + String(state->get_connection_source(i).simplified().c_escape()) + "\"";
+			connstr += " to=\"" + String(state->get_connection_target(i).simplified().c_escape()) + "\"";
+			connstr += " method=\"" + String(state->get_connection_method(i).c_escape()) + "\"";
```
Comment (dalexeev) [inline scene/resources/resource_format_text.cpp:2295]:
```suggestion
			connstr += " signal=\"" + String(state->get_connection_signal(i)).c_escape() + "\"";
			connstr += " from=\"" + String(state->get_connection_source(i).simplified()).c_escape() + "\"";
			connstr += " to=\"" + String(state->get_connection_target(i).simplified()).c_escape() + "\"";
			connstr += " method=\"" + String(state->get_connection_method(i)).c_escape() + "\"";
```

**Comment by LimestaX** [comment]
@dalexeev Woops, thanks for catching that, added to commit

**Comment by dalexeev** [comment]
Please [squash commits into one](https://docs.godotengine.org/en/stable/contributing/workflow/pr_workflow.html#the-interactive-rebase). To do this you need a local git client, GitHub web interface doesn't support it.

**Comment by LimestaX** [comment]
@dalexeev Okay, working on it.

**Comment by AThousandShips** [comment]
You missed the `git push --force` part

**Pull request closed without merging**